service/src/main/java/uk/gov/hmcts/reform/fpl/utils/JudgeAndLegalAdvisorHelper.java

```diff
@@ -53,6 +53,16 @@ public static void removeAllocatedJudgeProperties(JudgeAndLegalAdvisor judgeAndL
         judgeAndLegalAdvisor.setUseAllocatedJudge(null);
     }
 
+    public static JudgeAndLegalAdvisor prepareJudgeFields(JudgeAndLegalAdvisor judgeAndLegalAdvisor,
+                                                          Judge allocatedJudge) {
+        if (allocatedJudge.hasEqualJudgeFields(judgeAndLegalAdvisor)) {
+            return judgeAndLegalAdvisor.resetJudgeProperties();
+        } else {
+            judgeAndLegalAdvisor.setUseAllocatedJudge("No");
```

Comment:
Why do we explicitly need to set it to no? is ignoring it and leaving as null not enough?

Reply:
Because if the code reaches here then we know that there is a judge on the SDO, which means they must have selected something - if we left it as null we would lose the user's selection of 'No' so they would have to click it again